extlinks/links/management/commands/remove_ezproxy_collection.py

```diff
@@ -0,0 +1,155 @@
+
+    def handle(self, *args, **options):
+        ezproxy_org = self._get_ezproxy_organisation()
+        ezproxy_collection = self._get_ezproxy_collection()
+        url_patterns = self._get_ezproxy_url_patterns(ezproxy_collection)
+
+        if ezproxy_org and ezproxy_collection and url_patterns:
+            self._delete_aggregates_ezproxy(
+                ezproxy_org, ezproxy_collection, url_patterns
+            )
+
+        # Get LinkEvents that have no URLPatterns associated
+        linkevents = LinkEvent.objects.filter(url__isnull=True)
+
+        collections = Collection.objects.all()
+
+        self._process_linkevents_collections(linkevents, collections)
+
+    def _get_ezproxy_organisation(self):
+        """
+        Gets the EZProxy organisation, or returns None if it's already been deleted
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        Organisation object or None
+        """
+        if Organisation.objects.filter(name="Wikipedia Library OCLC EZProxy").exists():
+            return Organisation.objects.get(name="Wikipedia Library OCLC EZProxy")
+
+        return None
+
+    def _get_ezproxy_collection(self):
+        """
+        Gets the EZProxy collection, or returns None if it's already been deleted
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        Collection object or None
+        """
+        if Collection.objects.filter(name="EZProxy").exists():
+            return Collection.objects.get(name="EZProxy")
+
+        return None
+
+    def _get_ezproxy_url_patterns(self, collection):
+        """
+        Gets the EZProxy collection, or returns None if it's already been deleted
+
+        Parameters
+        ----------
+        collection: The collection the URLPatterns belong to
+
+        Returns
+        -------
+        URLPattern object or None
+        """
+        if collection and URLPattern.objects.filter(collection=collection).exists():
+            return URLPattern.objects.get(collection=collection)
+
+        return None
+
+    def _delete_aggregates_ezproxy(self, ezproxy_org, ezproxy_collection, url_patterns):
+        """
+        Deletes any aggregate with the EZProxy collection and organisation,
+        then deletes the collection, organisation and url patterns
+
+        Parameters
+        ----------
+        ezproxy_org: Organisation
+        The organisation to filter and delete the aggregates tables and that
+        will later be deleted
+
+        ezproxy_collection: Collection
+        The collection to filter and delete the aggregates tables and that
+        will later be deleted
+
+        url_patterns: URLPattern
+        The EZProxy URLPatterns that will be deleted
+
+        Returns
+        -------
+
+        """
+        LinkAggregate.objects.filter(
+            organisation=ezproxy_org, collection=ezproxy_collection
+        ).delete()
```

Comment:
Just double checking - are the cascade settings set so that this won't also inadvertently delete objects we want to keep, like the linkevents?

Reply:
Pretty sure cascading works in the other direction. The foreignkeys in the aggregates models are all set to cascade, which means if the referenced objects (linkevents, orgs, users) are deleted then the aggregates that reference them get deleted too. Deleting the aggregates should only impact models that had a forgeinkey reference to the aggregates, and we don't have any of those.